lib/codegen/CreateTerm.cpp

```diff
@@ -952,12 +952,11 @@ std::pair<llvm::Value *, bool> create_term::create_allocation(
               fn_name, constructor, false, true, false, location_stack),
           true);
     }
-    if (auto cat
-        = dynamic_cast<kore_composite_sort *>(symbol->get_arguments()[0].get())
-              ->get_category(definition_)
-              .cat;
+    if (auto *sort
+        = dynamic_cast<kore_composite_sort *>(symbol->get_arguments()[0].get());
         symbol_decl->attributes().contains(attribute_set::key::SortInjection)
-        && (cat == sort_category::Symbol)) {
+        && (sort->get_category(definition_).cat == sort_category::Symbol)
+        && definition_->get_supersorts()[sort].empty()) {
```

Comment:
This doesn't seem quite right to me. I'm honestly a little surprised that the tests are passing. If I understand correctly, `get_supersorts` is returning a map from each supersort to the set of subsorts associated with that sort. In such a case, the condition in this case ought to be negated, right? The if block should succeed only if the sorts are non-empty.

Am I misunderstanding the behavior of the code somehow? And if not, can you add some kind of test that would detect this bug?

Reply:
I think that what you're saying makes sense to me, Dwight. However, 10% of our tests (using lit) fail if I use the exact opposite condition. I'm blocked from understanding how this ought to work, then. For comparison, as it is implemented, it also passes all K regression tests, but the opposite condition also fails in some tests...